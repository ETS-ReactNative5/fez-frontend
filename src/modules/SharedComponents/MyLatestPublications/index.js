export {default as MyLatestPublications} from './containers/MyLatestPublications';
