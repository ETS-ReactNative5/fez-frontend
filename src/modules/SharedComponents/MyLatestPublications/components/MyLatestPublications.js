import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {withRouter} from 'react-router-dom';

import {routes} from 'config';
import {locale} from 'locale';

import {InlineLoader} from 'modules/SharedComponents/Toolbox/Loaders';
import {PublicationsList} from 'modules/SharedComponents/PublicationsList';
import RaisedButton from 'material-ui/RaisedButton';

export class MyLatestPublications extends PureComponent {
    static propTypes = {
        latestPublicationsList: PropTypes.array,
        totalPublicationsCount: PropTypes.number,
        isLoading: PropTypes.bool,
        history: PropTypes.object.isRequired
    };

    static defaultProps = {
        latestPublicationsList: [],
        totalPublicationsCount: null,
        isLoading: false
    };

    _viewYourResearch = () => {
        this.props.history.push(routes.pathConfig.records.mine);
    };

    render() {
        const txt = locale.components.myLatestPublications;

        if (this.props.isLoading) {
            return (
                <div className="isLoading is-centered">
                    <InlineLoader message={txt.loading}/>
                </div>
            );
        }

        return (
            <div  className="latestPubs">
                <PublicationsList
                    publicationsList={this.props.latestPublicationsList}
                    showDefaultActions/>
                <div className="columns">
                    <div className="column is-hidden-mobile"/>
                    <div className="column is-narrow">
                        <RaisedButton
                            secondary
                            label={`${txt.viewAllButtonLabel} (${this.props.totalPublicationsCount})`}
                            onTouchTap={this._viewYourResearch}/>
                    </div>
                </div>
            </div>
        );
    }
}

export default withRouter(MyLatestPublications);

