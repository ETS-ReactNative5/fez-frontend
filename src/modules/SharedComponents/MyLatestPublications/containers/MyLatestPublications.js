import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {withRouter} from 'react-router-dom';
import MyLatestPublications from '../components/MyLatestPublications';
import * as actions from 'actions';

const mapStateToProps = (state) => {
    return {
        ...state.get('myLatestPublicationsReducer'),
        accountAuthorDetailsLoading: state.get('accountReducer').accountAuthorDetailsLoading
    };
};

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    };
}

const MyTrendingPublicationsContainer = connect(mapStateToProps, mapDispatchToProps)(MyLatestPublications);

export default withRouter(MyTrendingPublicationsContainer);
