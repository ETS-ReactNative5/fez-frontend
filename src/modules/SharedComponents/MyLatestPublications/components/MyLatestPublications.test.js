import {latestPubsPayload} from 'mock/data/testing/latestPublications';
import MyLatestPublications from './MyLatestPublications';

function setup(testProps, isShallow = true){
    const props = {
        ...testProps
    };
    return getElement(MyLatestPublications, props, isShallow);
}

describe('Component MyLatestPublications', () => {
    it('should render latest publications', () => {
        const wrapper = setup({latestPublicationsList: latestPubsPayload.data, totalPublicationsCount: latestPubsPayload.total});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
