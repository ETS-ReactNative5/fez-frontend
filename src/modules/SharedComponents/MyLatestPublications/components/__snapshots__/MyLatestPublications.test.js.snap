// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MyLatestPublications should render latest publications 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`Component MyLatestPublications should render loading indicator 1`] = `
<Route
  render={[Function]}
/>
`;
